### PR TITLE
Fix --arch and --os flags to work correctly

### PR DESCRIPTION
**Pull request by rhatdan**
Currently podman implements --override-arch and --overide-os
But Podman has made these aliases for --arch and --os.  No
reason to have to specify --override, since it is clear what
the user intends.

Currently if the user specifies an --override-arch field but the
image was previously pulled for a different Arch, podman run uses
the different arch.  This PR also fixes this issue.

Fixes: https://github.com/containers/podman/issues/8001

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@lisongmin PTAL

File: cmd/podman/common/create.go
Comment (Luap99) [inline cmd/podman/common/create.go:483]: If I remember correctly you added autocomplete functions for `arch` and `os`. It would be good if you add them here.

Comment (rhatdan) [reply cmd/podman/common/create.go:483]: Thanks for reminding me.

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

**Comment by TomSweeneyRedHat** [comment]
LGTM 
assuming happy testst

**Comment by edsantiago** [comment]
LGTM but all I can give it right now is a cursory once-over. FWIW I did run my usual `ack override-` and `ack '\.Override'` (`ack` is a nice `grep -r`) to look for strays: none found. Nice.

**Review by saschagrunert — APPROVED** [review]
LGTM

**Comment by vrothberg** [comment]
/hold

I doubt that this change fixes https://github.com/containers/podman/issues/8001. Will test a bit.

File: cmd/podman/containers/create.go
Comment (vrothberg) [inline cmd/podman/containers/create.go:240]: @rhatdan can you add a `// FIXME` that states that all this logic should go into the backend? While this fix works for the clients, it does not work for the server.

I'd prefer something similar to https://github.com/vrothberg/libpod/commit/6aa9622350853aa34ec3423f0ae1dd1fd730532e but I never managed to get back to it.  I would love to remove the `pull` entirely (as the backend does that already) but that would lose the progress bars for the remote client.

**Comment by vrothberg** [comment]
> I doubt that this change fixes #8001. Will test a bit.

Looking good :+1:

File: cmd/podman/containers/create.go
Comment (rhatdan) [reply cmd/podman/containers/create.go:240]: When we do the containers/common image, could we take care of this there?

Comment (vrothberg) [reply cmd/podman/containers/create.go:240]: Yes we could do that. But that would be based on `libpod/image` which is currently not doing that it should. That's why I think that https://github.com/vrothberg/libpod/commit/6aa9622350853aa34ec3423f0ae1dd1fd730532e is closer to tackle the root of the problem.

**Pull request merged**
